tracer/src/Datadog.Trace/Iast/ReturnedHeadersAnalyzer.cs

```diff
@@ -38,25 +41,60 @@ internal static void Analyze(IHeaderDictionary responseHeaders, IntegrationId in
         }
 
         IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.XContentTypeHeaderMissing);
+        IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.HstsHeaderMissing);
         string contentTypeValue = responseHeaders[ContentType];
         string contentOptionValue = responseHeaders[XContentTypeOptions];
+        string strictTransportSecurityValue = responseHeaders[StrictTransportSecurity];
+        string xForwardedProtoValue = responseHeaders[XForwardedProto];
 
-        if (!IsHtmlResponse(contentTypeValue) || IsNoSniffContentOptions(contentOptionValue))
+        if (!IsHtmlResponse(contentTypeValue))
         {
             return;
         }
 
         LaunchXContentTypeOptionsVulnerability(integrationId, serviceName, contentTypeValue, contentOptionValue);
+        LaunchStrictTransportSecurity(integrationId, serviceName, strictTransportSecurityValue, xForwardedProtoValue, protocol);
     }
 
     private static void LaunchXContentTypeOptionsVulnerability(IntegrationId integrationId, string serviceName, string contentTypeValue, string contentOptionValue)
     {
-        if (!string.IsNullOrEmpty(contentTypeValue))
+        if (!string.IsNullOrEmpty(contentTypeValue) && !IsNoSniffContentOptions(contentOptionValue))
         {
             IastModule.OnXContentTypeOptionsHeaderMissing(integrationId, contentOptionValue, serviceName);
         }
     }
 
+    private static void LaunchStrictTransportSecurity(IntegrationId integrationId, string serviceName, string strictTransportSecurityValue, string xForwardedProtoValue, string protocol)
+    {
+        if (!string.Equals(protocol, "https", StringComparison.OrdinalIgnoreCase)
+            && !string.Equals(xForwardedProtoValue, "https", StringComparison.OrdinalIgnoreCase))
+        {
+            return;
+        }
+
+        if (!IsValidStrictTransportSecurityValue(strictTransportSecurityValue))
+        {
+            IastModule.OnStrictTransportSecurityHeaderMissing(integrationId, serviceName);
+        }
+    }
+
+    // Strict-Transport-Security has a valid value when it starts with max-age followed by a positive number (>0),
+    // it can finish there or continue with a semicolon ; and more content.
+    private static bool IsValidStrictTransportSecurityValue(string strictTransportSecurityValue)
+    {
+        if (string.IsNullOrEmpty(strictTransportSecurityValue) || !strictTransportSecurityValue.StartsWith(MaxAgeConst, StringComparison.OrdinalIgnoreCase))
+        {
+            return false;
+        }
+
+        var index = strictTransportSecurityValue.IndexOf(';');
+
+        var maxAge = (index >= 0 ? strictTransportSecurityValue.Substring(MaxAgeConst.Length, index - MaxAgeConst.Length) :
+            strictTransportSecurityValue.Substring(MaxAgeConst.Length));
+
+        return (int.TryParse(maxAge, out var maxAgeInt) && maxAgeInt > 0);
```

Comment:
minor: could this field be bigger than an int32? Could we parse as an unsigned number to avoid having to check if it's positive?

Reply:
That's interesting. The definition does not seem to establish a maximum number. In practice, it seems that some implementations rely on a 32 bits integer but I guess that we could use an ulong and consider it a right value if parsed.

More info: https://datatracker.ietf.org/doc/html/rfc6797#section-6.1.1

Reply:
Fixed!